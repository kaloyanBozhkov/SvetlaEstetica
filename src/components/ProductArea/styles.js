const styles = {
    productArea: {
        marginTop: '3rem',
        w: '100%',
        gap: '2rem',
        gridTemplateColumns: {
            xs: '1fr',
            md: 'repeat(2, 1fr)',
            base: 'repeat(4, 1fr)'
        },
        gridTemplateRows: '31rem'
    }
}

export default styles