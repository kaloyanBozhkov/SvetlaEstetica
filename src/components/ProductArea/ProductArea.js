import React from 'react'
import { Grid, PseudoBox } from '@chakra-ui/core'
import ProductCard from '../ProductCard/ProductCard'
import styles from './styles'

const ProductArea = ({ items = [], ...gridStyleProps}) => {
    return (
        <Grid {...styles.productArea} {...gridStyleProps}>
            {items.map((item, index) => (
                <PseudoBox {...styles.producetWrapper}>
                    <ProductCard {...item} key={index} />
                </PseudoBox>
            ))}
        </Grid>
    )
}

export default ProductArea