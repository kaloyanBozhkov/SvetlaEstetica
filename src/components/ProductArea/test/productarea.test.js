import { shallow } from 'enzyme'
import React from 'react'
import ProductArea from '../ProductArea'

describe('testing product area', () => {
    it('should render component', () => {
        const wrapper = shallow(<ProductArea />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should handle rendering items arr if provided', () => {
        const items = [
            {
                title: 'some product',
                price: '19.99',
                currency: '$',
                img: 'img src',
                description: 'some description'
            },
            {
                title: 'some other product',
                price: '18.99',
                currency: '$',
                img: 'img2 src',
                description: 'some other description'
            }
        ]
        const wrapper = shallow(<ProductArea items={items} />)
        const productCards = wrapper.find('ProductCard')
        expect(productCards.length).toEqual(items.length)
    })

    it('shoould render product area with passed style props', () => {
        const wrapper = shallow(<ProductArea gridTemplateColumns="1fr 1fr" />)
        expect(wrapper.find('Grid').prop('gridTemplateColumns')).toEqual('1fr 1fr')
    })
})