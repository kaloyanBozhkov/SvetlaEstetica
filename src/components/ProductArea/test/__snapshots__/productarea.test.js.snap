// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing product area should render component 1`] = `
<Grid
  autoRows="40rem"
  gap="4rem 2rem"
  gridTemplateColumns={
    Object {
      "base": "repeat(4, 1fr)",
      "lg": "repeat(5, 1fr)",
      "md": "repeat(2, 1fr)",
      "mdX": "repeat(3, 1fr)",
      "xs": "1fr",
    }
  }
  marginTop="3rem"
  w="100%"
/>
`;
